service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java

```diff
@@ -171,4 +146,41 @@ public Boolean sampleHasMetadataUpdates(SampleMetadata existingSampleMetadata,
         }
         return Boolean.FALSE;
     }
+
+    @Override
+    public PublishedMetadbSample getPublishedMetadbSample(UUID metadbSampleId) throws Exception {
+        MetadbSample sample = getMetadbSample(metadbSampleId);
+        return new PublishedMetadbSample(sample);
+    }
+
+    @Override
+    public List<PublishedMetadbSample> getPublishedMetadbSampleListByCmoPatientId(
+            String cmoPatientId) throws Exception {
+        List<SampleMetadata> sampleMetadataList = sampleRepository
+                .findAllSampleMetadataListByCmoPatientId(cmoPatientId);
+        List<PublishedMetadbSample> samples = new ArrayList<>();
+        for (SampleMetadata sample: sampleMetadataList) {
+            // TODO: update method to fill in sample details in a more inclusive
```

Comment:
still need this change here

Reply:
to be completed as part of [#480](https://app.zenhub.com/workspaces/metadb-scrum-5fcfa28a4ccc1600165347b6/issues/mskcc/cmo-metadb/480)